src/dndbeyond_character.pyj

```diff
@@ -66,12 +66,15 @@ def rollSavingThrow():
     rollAbilityOrSavingThrow("ct-ability-saving-throws-pane", "saving-throw")
 
 def rollInitiative():
+    nonlocal character, settings
     initiative = $(".ct-initiative-box__value").text()
     advantage = $(".ct-initiative-box__advantage").length > 0
     if initiative == "":
         initiative = $(".ct-combat-mobile__extra--initiative .ct-combat-mobile__extra-value").text()
         advantage = $(".ct-combat-mobile__advantage").length > 0
-    #console.log("Initiative " + ("with" if advantage else "without") + " advantage : " + initiative)
+    console.log(character)
+    initiative = initiative if not settings.getSetting("initiative-with-dex-score", False) else initiative

```

Comment:
Same as above, split the if into a non-tertiary operator.
Also, it should be `character.getGlobalSetting`, I think `settings` is only a dictionary, so `settings.getSetting` would not work.
In this case, you can also get the DEX abilities using : 
```python
   if character.getGlobalSetting("initiative-with-dex-score")
     tiebreaker = 0
     for ability in character._abilities:
        if ability[1] == "DEX":
           tiebreaker = ability[2]
```
And finally, remove the `console.log` lines

Reply:
Would you rather me fix this in both dndbeyond.pyj & dndbeyond_character.pyj (as is the current approach) or did you want me to have dndbeyond_character.pyj call the parent class method and pass along the initiative modifier?

Reply:
The code is honestly a mess, so there isn't really any 'parent method' in this case. The 'rollInitiative' in dndbeyond.pyj is a method that exists only for the `Monster` class (the only common functions are in `CharacterBase` which is the `getSetting` and `getGlobalSetting` I think), so the change has to go in both files/classes.

Reply:
Ah, OK. Is there any value in doing a rolling refactor such that we move this function into CharacterBase with the intention of fixing the others as you go? Or would you rather keep them the way they are so that you don't find yourself in a 1/2 way refactored state?

Reply:
A refactor at this point would be more of a rewrite. Statblocks and character sheets are also very different in D&D Beyond. A roll Initiative for a monster is about simply adding dexterity, but also modifying the statblock to add my own 'Roll Initiative' button for convenience.
For a character, it's about checking the initiative value in the sidebar, verifying if there are 'advantage on initiative' modifiers due to class features, etc.. 
The way the two work is too different to have anything merged. If you had a common function name for rolling initiative, you'd just have them override it both, so it's not worth pursuing at this time.

Note: Beyond20 was initially written as a thought experiment, a "would it be possible to... ?" question, from before I knew how to write javascript (explains the .pyj) and from before I knew html/css (I still don't). It worked so easily and quickly, and got so much traction that it quickly evolved, but the code is unfortunately still very messy. I apologize for that :)

Reply:
I think you've described the origin story for just about any project that's ever come to be. :)

No need for apologies - it works! That's a testament to itself!

Rewrites can be very good and very bad. If you actually have the time (and energy) you need to follow through, then it could become the new base and you could incorporate all you learned. However, its easier to iterate on working code then to hack on vaporware. Also, the longer you wait to rewrite, the more you'll have to write to achieve feature parity. Well-written software can often lose to worse written software that was good enough to work when people needed it.

You're the maintainer, so I obviously want to defer to your vision.

It might make sense to move towards a Model-View-Controller type organization. The abstraction I would make would be that characters and monsters are special cases of a creature, similar to how the game is structured. Creatures (models) would have abilities, speed, type, size, etc and corresponding (controller) methods to interact with them (e.g. rollInitiative). Your "views" would be the code that inject the buttons into the page. You would also need controllers to parse the dndbeyond page specifics.

I don't think the underlying tech you pick matters all that much once you have a portable/modular architecture.